src/index.js

```diff
@@ -44,24 +44,19 @@ module.exports = class BlockService {
   }
 
   putStream () {
-    let ps
     if (this.isOnline()) {
-      // NOTE: This will have to change in order for bitswap
-      // to understand CID
-      ps = this._bitswap.putStream()
+      return this._bitswap.putStream()
     } else {
-      ps = this._repo.blockstore.putStream()
+      return pull(
+        pull.map((blockAndCID) => {
```

Comment:
Btw, I would love to have something nicer, the `blockAndCID` isn't my favourite, but we really can't use something as `Block` right now unless we 'patch' it every time to have the right CID, which won't happen when reading from the repo, and there the only identifier is the multihash. Right now, `Block` is really just a buffer + a convenience method that invokes multihashing, it might make sense to do what @kumavis brought up in the last call and just remove that type and do `{ data: cid: }` objects all the way. (Still not convinced of what is simpler)